pkg/proxy/iptables/proxier_test.go

```diff
@@ -913,6 +913,50 @@ func TestNodePort(t *testing.T) {
 	}
 }
 
+func TestHealthCheckNodePort(t *testing.T) {
```

Comment:
hiya @hanlins , looking at this unit test,  do we also need a `sig-network`  E2E for this? what E2Es break if this is not working ?

Reply:
Oh true. I was thinking about it early today but don't know if there's a way to change the e2e VM settings? My concern is if I just modify the `INPUT` chain policy to `DROP` then we might lose ssh connection to that VM instantly and failed the test (and lose the VMs, maybe other risks). Also I would say if we change the chain policy to `DROP` then local pod to pod VM will break since there's no rule to allow traffic from/to `lo` interface. Personally that seems to be a bigger topic than just focusing on fixing the healthcheck node port issue, maybe we can create new issues to track them? Maybe one for adding a rule to enable traffic to/from `lo` interface, and the other to address the e2e test coverage, what's ur take? 😆

Reply:
no need to make an e2e for it, i guess all that matters is... knowing which of the `sig-network` e2e's we can use to detect this regression?  

i bet  theres something that requires nodeport healthchecks (i.e. maybe in test/e2e/network/services.go) which demonstrates this bug ?

Reply:
We have quite a few e2es already on health check node port because it's required for any Service LB w/ externalTrafficPolicy=Local. The tricky part is testing this on a node using default `INPUT` policy of `DROP`.

Reply:
> My concern is if I just modify the INPUT chain policy to DROP then we might lose ssh connection to that VM instantly

You don't need to change the default `INPUT` chain policy, you can just do

    iptables -A INPUT -p tcp --dport <PORT> -j DROP

or whatever

Reply:
> iptables -A INPUT -p tcp --dport <PORT> -j DROP

This rule would conflict with the rules we add from kube-proxy though.

Reply:
> This rule would conflict with the rules we add from kube-proxy though.

That's the goal! Add a rule (to the very end of the `INPUT` chain) that intentionally drops packets directed just at this health check port. (When you quoted my reply the "PORT" was missing, but the rule as I originally typed it was supposed to apply only to that particular part.) The new `-j ACCEPT` rule added by this PR should run before that rule gets hit, rendering it irrelevant.

Reply:
That depends on if we prepend or append though right? Do we always prepend filter rules?

Reply:
The actual per-service rules get added to kubernetes-specific chains, not to the top-level `INPUT`, `OUTPUT`, etc, chains. We should be able to arrange for a single rule added by a test case to be added to the right place relative to the `-j KUBE-WHATEVER` rule in `INPUT`.